File: 미분류/10828.py
import sys
input = sys.stdin.readline

N = int(input())
stack = []
size = 0
for _ in range(N):
    control = input().split()
    if control[0] == "push":
        stack.append(int(control[1]))
        size += 1

    elif control[0] == "top":
        if size == 0:
            print(-1)
        else:
            print(stack[-1])

    elif control[0] == "pop":
        if size == 0:
            print(-1)
        else:
            print(stack[-1])
            stack.pop()
            size -= 1

    elif control[0] == "empty":
        if size == 0:
            print(1)
        else:
            print(0)
    else:
        print(size)

File: 미분류/10828.c
#include <stdio.h>
#include <string.h>

typedef struct ArrayStack {
    int top;
    int stack[10000];
}ArrayStack;

void initStack (ArrayStack* AS){
    AS->top = -1;
}

void pushStack(ArrayStack* AS, int data){
    AS->stack[++AS->top] = data;
}

int popStack(ArrayStack* AS){
    if (AS->top == -1){
        return -1;
    }
    return AS->stack[AS->top--];
}

int sizeStack(ArrayStack* AS){
    return AS->top + 1;
}

int topStack(ArrayStack* AS){
    if (AS->top == -1){
        return -1;
    }
    return AS->stack[AS->top];
}

int emptyStack(ArrayStack* AS){
    return AS->top == -1;
}

int main () {
    int N, temp;
    ArrayStack AS;
    char control[6] = "";
    scanf("%d", &N);
    initStack(&AS);

    while(N--){
        memset(control, 0, sizeof(control));
        scanf("%s", control);

        if(!strcmp("push", control)){
            scanf("%d", &temp);
            pushStack(&AS, temp);
        }
        else if (!strcmp("top", control)){
            printf("%d\n", topStack(&AS));
        }
        else if (!strcmp("size", control)){
            printf("%d\n", sizeStack(&AS));
        }
        else if (!strcmp("empty", control)){
            printf("%d\n", emptyStack(&AS));
        }
        else{
            printf("%d\n", popStack(&AS));
        }
    }
}